src/includes_normalize-win32.cc

```diff
@@ -28,16 +28,26 @@ namespace {
 
 bool InternalGetFullPathName(const StringPiece& file_name, char* buffer,
                              size_t buffer_length, string *err) {
-  DWORD result_size = GetFullPathNameA(file_name.AsString().c_str(),
-                                       buffer_length, buffer, NULL);
+  wchar_t w_buffer[_MAX_PATH];
+  wstring w_file_name = Utf8ToWide(file_name.AsString());
+  DWORD result_size =
+      GetFullPathNameW(w_file_name.c_str(), buffer_length, w_buffer, NULL);
   if (result_size == 0) {
-    *err = "GetFullPathNameA(" + file_name.AsString() + "): " +
-        GetLastErrorString();
+    *err = "GetFullPathNameW(" + file_name.AsString() +
+           "): " + GetLastErrorString();
     return false;
   } else if (result_size > buffer_length) {
     *err = "path too long";
     return false;
   }
+
+  string s_buffer = WideToUtf8(w_buffer);
```

Comment:
1. std::
2. Another place where you're allocating a new string, then copying it's guts out, then throwing away the string you allocated.

WideToUtf8 needs to be modified so that you can pass in the destination buffer.

Or you need to use the guts of WideToUtf8 here directly, so you're not wastefully allocating and then throwing away.